pkg/model/api/v1/dashboard.go

```diff
@@ -142,6 +142,43 @@ func (d *DashboardSpec) validate() error {
 	return nil
 }
 
+// verifyAndSetJSONRef will check that each JSON Reference are pointing to an existing object and will set the related pointer in the JSONRef.Object
+func (d *DashboardSpec) verifyAndSetJSONReferences() error {
+	for _, layout := range d.Layouts {
+		switch spec := layout.Spec.(type) {
+		case *dashboard.GridLayoutSpec:
+			for _, item := range spec.Items {
+				if err := d.checkAndSetRef(item.Content); err != nil {
+					return err
+				}
+			}
+
+		}
+	}
+	return nil
+}
+
+func (d *DashboardSpec) checkAndSetRef(ref *common.JSONRef) error {
+	// ref.Path should be like [ "spec", "panels", <name> ]
+	var panelsRefPath = []string{"spec", "panels"}
```

Comment:
at this level, there is no way to be sure that the complete path is starting by `spec`. It could start by something else or have a different depth